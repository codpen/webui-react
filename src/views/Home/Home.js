import React from 'react';
import {Col, Row} from "reactstrap";
import BackendStatusCard from "../Base/BackendStatusCard/BackendStatusCard";
import "../../utils/Global";
import RunningJobs from "../Base/RunningJobs";

class Home extends React.Component {


    render() {
        return (<div>
            <h2>Welcome to Rclone dashboard. </h2>
            <p>Begin by creating a new remote config from the left sidebar.</p>
            <Row>
                <Col lg={4} sm={12}>
                    <BackendStatusCard ipAddress={global.ipAddress} mode={"card"}/>
                </Col>
                {/*<Col lg={4} sm={12}>*/}
                {/*    <BandwidthStatusCard/>*/}
                {/*</Col>*/}
            </Row>
            <h2>Jobs</h2>

            <RunningJobs mode={"full-status"}/>

        </div>);
    }
}

export default Home;
