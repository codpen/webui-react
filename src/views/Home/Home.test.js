import React from "react";
import {shallow} from "enzyme";
import {findByTestAtrr, testStore} from "../../../Utils";
import Home from "./Home";
import toJson from "enzyme-to-json";

const setUp = (intialState = {}, props = {}) => {
    const store = testStore(intialState);
    return shallow(<Home {...props} store={store}/>);
};


describe('Home Component', function () {


    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const initialState = {};

            const props = {};
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            const component = findByTestAtrr(wrapper, "homeComponent");
            expect(component).toHaveLength(1);
        });

        it('should match snapshot', function () {
            expect(toJson(wrapper)).toMatchSnapshot()
        });
    });

});