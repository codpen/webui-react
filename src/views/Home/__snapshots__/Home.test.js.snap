// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders should match snapshot 1`] = `
<div
  data-test="homeComponent"
>
  <h2>
    Welcome to Rclone dashboard. 
  </h2>
  <p>
    Begin by creating a new remote config from the left sidebar.
  </p>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      lg={4}
      sm={12}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Connect(BackendStatusCard)
        ipAddress={null}
        mode="card"
      />
    </Col>
    <Col
      lg={4}
      sm={12}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Connect(BandwidthStatusCard) />
    </Col>
  </Row>
  <h2>
    Jobs
  </h2>
  <Connect(RunningJobs)
    mode="full-status"
  />
</div>
`;
